### PR TITLE
[Solved #15] Communication protocol design documentation

### LINKED ISSUE #15: Application communication protocol design
Epic:

### Description
Design and create documentation of front-end and back-end communication protocol.

### Tasks

- [x] Study communication protocol and sockets
- [x] Authentication tech solution
- [x] Event communication protocol documentation

### PR DESCRIPTION
**Pull request by supunTE**
Solved #15

Preview - 
https://github.com/supunTE/razor/blob/feature/15-protocol-design/docs/communication/index.md

### CODE REVIEW

**Comment by supunTE** [comment]
**WIP**
- [x] Interface names in docs should change after refactoring the socket data models

Converting to draft

File: docs/communication/protocol/clear-player.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[<- Back](../index.md)
+
+# Clearing(Removing) Player (Player disconnect)
```
Comment (pasinducw) [inline docs/communication/protocol/clear-player.md:3]: Here, we need to wait a while before removing the player, as they might join right again with a new socket connection right?

File: docs/communication/protocol/end-race.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
+[<- Back](../index.md)
+
+# Ending Race (By timeout or Force end)
```
Comment (pasinducw) [inline docs/communication/protocol/end-race.md:3]: Please don't add dark colored boxes. Makes it difficult to read the text within
image

File: docs/communication/protocol/join-tournament.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
+[<- Back](../index.md)
+
+## Player Joining room
```
Comment (pasinducw) [inline docs/communication/protocol/join-tournament.md:3]: Please don't add dark colored boxes, makes it difficult to read the text
image

**Review by pasinducw — CHANGES_REQUESTED** [review]
Can you address the comments I left?

File: docs/communication/protocol/end-race.md (hunk 2 above)
Comment (supunTE) [reply docs/communication/protocol/end-race.md:3]: Solved 041a145f872847b2d7afc7c52dbced6ad8ed4577

File: docs/communication/protocol/join-tournament.md (hunk 3 above)
Comment (supunTE) [reply docs/communication/protocol/join-tournament.md:3]: Solved 041a145f872847b2d7afc7c52dbced6ad8ed4577

File: docs/communication/protocol/clear-player.md (hunk 1 above)
Comment (supunTE) [reply docs/communication/protocol/clear-player.md:3]: Add wait before removing the player - 603dcfafe557e403943dcafc2d76395524112d4a

**Review by pasinducw — APPROVED** [review]

**Review by dalanad — APPROVED** [review]

**Pull request merged**